stl/inc/optional

```diff
@@ -53,10 +53,14 @@ protected:
     _THROW(bad_optional_access{});
 }
 
+struct _Nontrivial_Dummy_type {
+    constexpr _Nontrivial_Dummy_type() noexcept {}
+};
+
 template <class _Ty, bool = is_trivially_destructible_v<_Ty>>
 struct _Optional_destruct_base { // either contains a value of _Ty or is empty (trivial destructor)
     union {
-        char _Dummy;
+        _Nontrivial_Dummy_type _Dummy;

```

Comment:
This is all well and good with the native ABIs, but we need to validate that we can sneak this type change past the class metadata when linking new-and-old CLR TUs. (And on line 82.) If not, we'll have to put off this part of the change to the next ABI breaking release and continue zeroing one byte.

Reply:
IIRC, only `/clr:pure` complains about "metadata mismatch", not plain `/clr`. Should validate though.

Reply:
Verified that before/after pure objects link, and that all combinations of clr/native before/after objects link.